torch/_inductor/runtime/benchmarking.py

```diff
@@ -0,0 +1,944 @@
+                log.debug("Early ranking is disabled, continuing benchmarking.")
+
+        callable_to_timing_ms = {}
+        for _callable, estimated_timing_ms in zip(callables, estimated_timings_ms):
+            callable_to_timing_ms[_callable] = estimated_timing_ms
+
+        if pruning_key is not None:
+            if should_enable_early_pruning():
+                cpu_launch_overhead_ms_per_iter_per_callable = (
+                    cpu_launch_overhead_ms_per_iter / len(callables)
+                )
+                target_timing_ms = min(estimated_timings_ms) * pruning_factor
+                callables_to_benchmark: List[Callable[[], Any]] = []
+                for _callable, timing_ms in sorted(
+                    callable_to_timing_ms.items(), key=lambda x: x[1]
+                ):
+                    if len(callables_to_benchmark) == pruning_limit:
+                        break
+                    if timing_ms <= target_timing_ms:
+                        callables_to_benchmark.append(_callable)
+                counters["inductor"]["benchmarking_early_pruning"] += len(
+                    callables
+                ) - len(callables_to_benchmark)
+                log.debug(
+                    "Early pruning pruned %d from %d total callables, continuing benchmarking with remaining %d callables.",
+                    len(callables) - len(callables_to_benchmark),
+                    len(callables),
+                    len(callables_to_benchmark),
+                )
+                cpu_launch_overhead_ms_per_iter = (
+                    cpu_launch_overhead_ms_per_iter_per_callable
+                    * len(callables_to_benchmark)
+                )
+                estimated_timings_ms = [
+                    callable_to_timing_ms[_callable]
+                    for _callable in callables_to_benchmark
+                ]
+            else:
+                log.debug("Early pruning is disabled, continuing benchmarking.")
+                callables_to_benchmark = callables
+        else:
+            callables_to_benchmark = callables
+
+        # see benchmark_gpu for details
+        benchmark_iters = self.reduce_benchmark_iters_to_max_benchmark_duration_ms(
+            memory_warmup_iters,
+            benchmark_iters,
+            cpu_launch_overhead_ms_per_iter,
+            max(estimated_timings_ms),
+            len(callables_to_benchmark),
+            max_benchmark_duration_ms,
+        )
+        benchmark_blocks = (
+            (benchmark_iters * 5 * len(callables_to_benchmark))
+            // (self.gpu_queue_limit - 1)
+        ) + 1
+        benchmark_iters_per_block = benchmark_iters // benchmark_blocks
+        memory_warmup_iters_per_block = memory_warmup_iters // benchmark_blocks
+        required_gpu_sleep_cycles_per_block = self.get_required_gpu_sleep_cycles(
+            memory_warmup_iters_per_block,
+            benchmark_iters_per_block,
+            cpu_launch_overhead_ms_per_iter,
+        )
```

Comment:
The reason for `5` is commented above but not clear here. Could we factor out some of this ?